phpunit/functional/SLMTest.php

```diff
@@ -1587,6 +1590,309 @@ public function testLaChange(): void
+        // action : ticket priority -> 4
+        $this->createItem(\SlaLevelAction::class, ['action_type' => 'assign', 'field' => 'priority', 'value' => 4, 'slalevels_id' => $level_1->getID()]);
+
+        $level_2 = $this->createItem(\SlaLevel::class, [
+            'name' => 'SLA level ' . time(),
+            'slas_id' => $sla->getID(),
+            'execution_time' => -30 * MINUTE_TIMESTAMP, // 30 minutes before TTR, 90 minutes elapsed
+            'is_active' => 1,
+            'is_recursive' => 1,
+            'match' => 'AND',
+        ]);
+        // criteria : ticket priority = 4
+        $this->createItem(\SlaLevelCriteria::class, ['criteria' => 'priority', 'condition' => 0, 'pattern' => '4', 'slalevels_id' => $level_2->getID()]);
+        // action : ticket priority -> 5
+        $this->createItem(\SlaLevelAction::class, ['action_type' => 'assign', 'field' => 'priority', 'value' => 5, 'slalevels_id' => $level_2->getID()]);
+
+        // --- 10:00 - create ticket and affect slalevels_id_ttr
+        $this->setCurrentTime('10:00:00');
+        $this->runSlaCron(); // no changes will be triggered
+        $ticket = $this->createTicket([
+            'status' => \CommonITILObject::INCOMING,
+            'slas_id_ttr' => $sla->getID(),
+            'priority' => 3,  // to match level_1 criteria
+        ]);
+        $ticket_id = $ticket->getID();
+
+        $this->assertEquals($sla->getID(), $ticket->fields['slas_id_ttr']);
+        // slalevels_id_ttr takes the first level, no matter what state or elapsed time is.
+        $this->assertEquals($level_1->getID(), $ticket->fields['slalevels_id_ttr']);
+        // TTR is computed from the time of creation 10:00 + 120 minutes -> 12:00
+        $this->assertEquals('12:00:00', substr($ticket->fields['time_to_resolve'], -8));
+        // next level to be processed is level_1
+        $this->assertTrue((new \SlaLevel_Ticket())->getFromDBByCrit(['tickets_id' => $ticket->getID(), 'slalevels_id' => $level_1->getID()]));
+
+        // --- 11:01 ticket : level 1 is reached
+        $this->setCurrentTime('11:01:00');
+        $this->runSlaCron();
+        $this->assertEquals($level_1->getID(), $ticket->fields['slalevels_id_ttr']); // not a relevant change, this won't change for the rest, kind of a bug ?
+        // next level to be processed is level_2
+        $this->assertTrue((new \SlaLevel_Ticket())->getFromDBByCrit(['tickets_id' => $ticket->getID(), 'slalevels_id' => $level_2->getID()]));
+        $ticket = new \Ticket();
+        $ticket->getFromDB($ticket_id);
+        $this->assertEquals(4, $ticket->fields['priority']); // level_1 action is applied
+
+        // --- 11:31 ticket : level 2 is reached
+        $this->setCurrentTime('11:31:00');
+        $this->runSlaCron();
+        // $this->assertEquals($level_2->getID(), $ticket->fields['slalevels_id_ttr']) // as noted above, this field does not change
```

Comment:
This is pretty weird. Is this field really used? Could we remove it from the database, or fix it to reflect the actual level applied?

Reply:
It's weird but I'm not sure it is used by plugins or other third party api. 
Removing the field from the code comes with some changes that requires to be very careful.
I've added this point to my todolist.